src/cart/crt_context.c

```diff
@@ -229,6 +229,8 @@ crt_context_provider_create(crt_context_t *crt_ctx, crt_provider_t provider, boo
 	if (ctx_idx < 0 || ctx_idx >= max_ctx_num) {
 		D_WARN("Provider: %d; Context limit (%d) reached\n",
 		       provider, max_ctx_num);
+		if (ctx_idx >=0)
+			crt_provider_put_ctx_idx(primary, provider, ctx_idx);
```

Comment:
This does not look correct to me. if we hit this case, means ctx_idx was >= max_ctx_num, in other words it was invalid.

We dont need to put invalid context back, and doing so today in fact might cause a crash as we access array at the invalid index.

603 void
 604 crt_provider_put_ctx_idx(bool primary, int provider, int idx)
 605 {
 606         struct crt_prov_gdata *prov_data = crt_get_prov_gdata(primary, provider);
 607
 608         if (prov_data->cpg_used_idx[idx] == false) {